advanced-formula-environment/examples/SpreadsheetLisp.ts

```diff
@@ -0,0 +1,1213 @@
+/*
+
+Spreadsheet Lisp (SL) v0.7.0
+
+Forged in the Advanced Formula Environment (AFE),
+courtesy of the minds on the Excel Labs team of Microsoft Garage.
+
+All design decisions codified herein are subject to emendation,
+useful or otherwise, following sufficient argumentation.
+No rights reserved.
+
+SL bows respectfully to its ancestral list of LISPs:
+(1.5 Common Scheme Racket Pico et.al.)
+
+DISCLAIMER:
+Most SL functions do not require macros to be enabled. [*.xlsx]
+The coolest SL functions require macros to be enabled. [*.xlsm]
+You are thus free, within the bounds of determinism, to choose your path forward.
+
+"For a dreamer is one who can only find his way by moonlight,
+ and his punishment is that he sees the dawn before the rest of the world."
+ - Oscar Wilde
+
+*/
+
+TABLE_OF_CONTENTS =
+    VLIST(
+        "Table of Contents",
+        "Types",
+        "Functions [.xlsm]",
+        "Functions [.xlsx]",
+        "Arcade",
+        "Epilogue");
+
+/*
+Table of Contents
+
+Types:
+Booleans
+Atoms
+
+Macro-dependent functions [*.xlsm]:
```

Comment:
Are the macro-dependent functions used by the macro functions, or is there a clean split between the two?

My preference would be to split them into into different files, so we can add a clear disclaimer for the macro-dependent ones, and then not worry about all the others.